src/bitcoin-miner.cpp

```diff
@@ -76,10 +76,6 @@ bool static ScanHash(const CBlockHeader *pblock, uint32_t &nNonce, uint256 *phas
         // caller will check if it has enough to reach the target
         if (((uint16_t *)phash)[15] == 0)
             return true;
-
-        // If nothing found after trying for a while, return -1
-        if ((nNonce & 0xfff) == 0)
```

Comment:
Note to anyone reading this: This check was not needed.  `ScanHash` would return here only to be immediately called again and continue from where it last left off.  So that's why it was removed.  I had to re-read `CpuMineBlockHasher` a few times to convince myself this is indeed not needed (I was trying to understand the rationale for it being here).